src/site/content/en/blog/use-long-term-caching/index.md

```diff
@@ -0,0 +1,783 @@
+
+
+After these changes, each build will generate two files instead of one: `main.[chunkhash].js` and
+`vendor.[chunkhash].js` (`vendors~main.[chunkhash].js` for webpack 4). In case of webpack 4,
+the vendor bundle might not be generated if dependencies are small – and that’s fine:
+
+```shell
+    $ webpack
+    Hash: ac01483e8fec1fa70676
+    Version: webpack 3.8.1
+    Time: 3816ms
+                            Asset   Size  Chunks             Chunk Names
+    ./main.00bab6fd3100008a42b0.js  82 kB       0  [emitted]  main
+    ./vendor.d9e134771799ecdf9483.js  47 kB       1  [emitted]  vendor
+```
+
+The browser would cache these files separately – and redownload only code that changes.
+
+### Webpack runtime code
+
+Unfortunately, extracting just the vendor code is not enough. If you try to
+change something in the app code:
+
+```js
+// index.js
+…
+…
+
+// E.g. add this:
+console.log('Wat');
+```
+
+you’ll notice that the `vendor` hash also changes:
+
+```shell
+                           Asset   Size  Chunks             Chunk Names
+./vendor.d9e134771799ecdf9483.js  47 kB       1  [emitted]  vendor
+```
+
+↓
+
+```shell
+                            Asset   Size  Chunks             Chunk Names
+./vendor.e6ea4504d61a1cc1c60b.js  47 kB       1  [emitted]  vendor
+```
+
+This happens because the webpack bundle, apart from the code of modules, has [_a
+runtime_](https://webpack.js.org/concepts/manifest/) – a small piece of code
+that manages the module execution. When you split the code into multiple files,
+this piece of code starts including a mapping between chunk ids and
+corresponding files:
+
+```js
+// vendor.e6ea4504d61a1cc1c60b.js
+script.src = __webpack_require__.p + chunkId + "." + {
+    "0": "2f2269c7f0a55a5c1871"
+}[chunkId] + ".js";
+```
+
+Webpack includes this runtime into the last generated chunk, which is `vendor`
+in our case. And every time any chunk changes, this piece of code changes too,
+causing the whole `vendor` chunk to change.
+
+To solve this, let’s move the runtime into a separate file. **In webpack 4,** this is
+achieved by enabling the `optimization.runtimeChunk` option:
+
+```js
+// webpack.config.js (for webpack 4)
+module.exports = {
+    optimization: {
+    runtimeChunk: true,
+    },
+};
+```
+
+**In webpack 3,** do this by creating an extra empty chunk with the `CommonsChunkPlugin`:
+
+```js
+// webpack.config.js (for webpack 3)
+module.exports = {
+    plugins: [
+    new webpack.optimize.CommonsChunkPlugin({
+        name: 'vendor',
+
+        minChunks: module => module.context &&
+        module.context.includes('node_modules'),
+    }),
+
+    // This plugin must come after the vendor one (because webpack
+    // includes runtime into the last chunk)
+    new webpack.optimize.CommonsChunkPlugin({
+        name: 'runtime',
+
+        // minChunks: Infinity means that no app modules
+        // will be included into this chunk
+        minChunks: Infinity,
+    }),
+    ],
+};
```

Comment:
```suggestion
module.exports = {
  plugins: [
    new webpack.optimize.CommonsChunkPlugin({
      name: 'vendor',
      minChunks: module => module.context && module.context.includes('node_modules')
    }),
    // This plugin must come after the vendor one (because webpack
    // includes runtime into the last chunk)
    new webpack.optimize.CommonsChunkPlugin({
      name: 'runtime',
      // minChunks: Infinity means that no app modules
      // will be included into this chunk
      minChunks: Infinity
    })
  ]
};
```